csrc/fusion_segmenter.cpp

```diff
@@ -1487,8 +1487,7 @@ void convertInputRfactorsToRoots(Fusion* fusion) {
     }
 
     NVF_ERROR(new_root_domain.size() == tv->domain()->contiguity().size());
-    auto new_td = IrBuilder::create<TensorDomain>(
-        new_root_domain, tv->domain()->contiguity());
```

Comment:
So this is where the issue happened.

For tensor with `allocation domain`, we expect contiguity flag to match that, but not with `root domain`. so this line would trigger an assert.
Even though in the code logic below, we are updating allocation domain with `new_td->setAllocationDomain(new_alloc_domain, new_td->contiguity());`

Switching to the new TensorDomain constructor allows us to set root domain and allocation domain at the same time to avoid the error.